File: solutions/344.reverse-string/reverse-string.java
class Solution {
    public String reverseString(String s) {
        char[] chars = s.toCharArray();
        int start = 0, end = chars.length -1;
        while(start < end){
            char temp = chars[end];
            chars[end] = chars[start];
            chars[start] = temp;
            start++;
            end--;
        }
        return new String(chars);
    }
}

File: solutions/344.reverse-string/reverse-string.js
/**
 * @param {character[]} s
 * @return {void} Do not return anything, modify s in-place instead.
 */
var reverseString = function(s) {
    let start = 0, end = s.length - 1;
    while(start < end){
        const temp = s[start];
        s[start++] = s[end];
        s[end--] = temp;
    }
};